CommonTools/UtilAlgos/interface/Merger.h

```diff
@@ -34,19 +36,20 @@ class Merger : public edm::global::EDProducer<> {
   explicit Merger(const edm::ParameterSet&);
   /// destructor
   ~Merger() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
```

Comment:
I'm completely guessing, but I thought it would be enough to call `addWithDefaultLabel` from the `fillDescriptions` of the template class (here, `Merger`), and then different `cfi`s would be generated for the different instances automatically. Something like this seemed to work in https://github.com/cms-sw/cmssw/pull/37039. (but again, I'm only guessing)

Reply:
That was my first attempt, but I got 
```
----- Begin Fatal Exception 17-Jan-2024 12:20:25 CET-----------------------
An exception of category 'LogicError' occurred while
   [0] Running executable "edmWriteConfigs"
   [1] Processing library /afs/cern.ch/work/s/sdonato/CMSSW_14_0_X_2024-01-16-2300/tmp/el8_amd64_gcc12/src/CommonTools/RecoAlgos/plugins/CommonToolsRecoAlgos_plugins/libCommonToolsRecoAlgos_plugins.so
   [2] Writing cfi files using ParameterSetDescriptions for module of base type EDProducer with plugin name 'SuperClusterMerger'
   [3] Executing function ConfigurationDescriptions::writeCfiForLabel
Exception Message:
Two cfi files are being generated with the same name in the same directory.
The cfi file name is 'simpleMergedTracks_cfi.py' and
the module label is 'simpleMergedTracks'.
This error is probably caused by an error in one or more fillDescriptions functions
where duplicate module labels are being passed to the ConfigurationDescriptions::add
function. All such module labels must be unique within a package.
If you do not want the generated cfi file and do not need more than one
description for a plugin, then a way to fix this is to use the addDefault
function instead of the add function.
There are 3 common ways this problem can happen.
1. This can happen when a module label is explicitly duplicated in one or more
fillDescriptions functions. Fix these by changing the module labels to be unique.
2. This can also happen when a module class is a template class and plugins are
defined by instantiations with differing template parameters and these plugins
share the same fillDescriptions function. Fix these by specializing the fillDescriptions
function for each template instantiation.
3. This can also happen when there is an inheritance heirarchy and multiple plugin modules
are defined using derived classes and the base class which share the same fillDescriptions
function. Fix these by redefining the fillDescriptions function in each derived class.
----- End Fatal Exception -------------------------------------------------
``` (see above).
But probably it is because I used `add` instead of `addWithDefaultLabel`

Reply:
Indeed using `addWithDefaultLabel` seems to solve the problem. Moved to #43731